platform-sdk/swirlds-config-api/src/main/java/com/swirlds/config/api/Configuration.java

```diff
@@ -42,134 +45,162 @@ public interface Configuration {
+     *                                  or the given type
      */
     List<String> getValues(String propertyName, List<String> defaultValue);
 
     /**
      * Returns a {@link List} of elements of the property with the given name
      *
-     * @param propertyName
-     * 		the name of the property
-     * @param propertyType
-     * 		the type of the elements
-     * @param <T>
-     * 		the generic type of the elements
+     * @param propertyName the name of the property
+     * @param propertyType the type of the elements
+     * @param <T>          the generic type of the elements
      * @return a {@link List} of elements of the property with the given name
-     * @throws NoSuchElementException
-     * 		if the property does not exist.
-     * @throws IllegalArgumentException
-     * 		if the raw {@link String} value of the property can not be converted to a list or the given type
+     * @throws NoSuchElementException   if the property does not exist.
+     * @throws IllegalArgumentException if the raw {@link String} value of the property can not be converted to a list
+     *                                  or the given type
      */
     <T> List<T> getValues(String propertyName, Class<T> propertyType)
             throws NoSuchElementException, IllegalArgumentException;
 
     /**
      * Returns a {@link List} of elements of the property with the given name or the given default {@link List}
      *
-     * @param propertyName
-     * 		the name of the property
-     * @param propertyType
-     * 		the type of the elements
-     * @param defaultValue
-     * 		the default {@link List}
-     * @param <T>
-     * 		the generic type of the elements
+     * @param propertyName the name of the property
+     * @param propertyType the type of the elements
+     * @param defaultValue the default {@link List}
+     * @param <T>          the generic type of the elements
      * @return a {@link List} of elements of the property with the given name or the given default list
-     * @throws IllegalArgumentException
-     * 		if the raw {@link String} value of the property can not be converted to a list or the given type
+     * @throws IllegalArgumentException if the raw {@link String} value of the property can not be converted to a list
+     *                                  or the given type
      */
     <T> List<T> getValues(String propertyName, Class<T> propertyType, List<T> defaultValue)
             throws IllegalArgumentException;
 
+    /**
+     * Returns a {@link Set} of string elements of the property with the given name
+     *
+     * @param propertyName the name of the property
+     * @return a {@link Set} of elements of the property with the given name
+     * @throws NoSuchElementException   if the property does not exist.
+     * @throws IllegalArgumentException if the raw {@link String} value of the property can not be converted to a set or
+     *                                  the given type
+     */
+    @Nullable
```

Comment:
Why are these nullable?  Isn't it nicer to have non-null?  Returning an empty set is free (https://github.com/openjdk/jdk17/blob/master/src/java.base/share/classes/java/util/Set.java#L455)

Reply:
Technically a converter that creates a native value out of a raw string can return null. That is a point in the Config API that is not really solved in a good way and needs some more work. But today it is exactly like that